pkg/gameserverallocations/controller.go

```diff
@@ -398,26 +398,32 @@ func (c *Controller) allocateFromRemoteCluster(gsa v1alpha1.GameServerAllocation
 	}
 
 	// TODO: Retry on transient error --> response.StatusCode >= 500
-	response, err := client.Post(connectionInfo.AllocationEndpoint, "application/json", bytes.NewBuffer(body))
-	if err != nil {
-		return nil, err
-	}
-	defer response.Body.Close() // nolint: errcheck
+	for i, endpoint := range connectionInfo.AllocationEndpoints {
+		response, err := client.Post(endpoint, "application/json", bytes.NewBuffer(body))
+		if err != nil {
+			return nil, err
+		}
+		defer response.Body.Close() // nolint: errcheck
 
-	data, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return nil, err
-	}
-	if response.StatusCode >= 400 {
-		// For error responses return the body without deserializing to an object.
-		return nil, errors.New(string(data))
-	}
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return nil, err
+		}
+		if response.StatusCode >= 500 && (i+1) < len(connectionInfo.AllocationEndpoints) {
+			// If there is a server error try a different endpoint
+			continue
+		}
+		if response.StatusCode >= 400 {
```

Comment:
Curious - why stop at a 400, why not continue to the next endpoint like a 500?

Reply:
4xx means there is an error from client. We only retry with new endpoint if there is an error from server e.g. 5xx.

Reply:
My concern would be, that if after I set everything up, and all is well - if something weird happens to one of my clusters, such that they return a 404, that could potentially stop all/a large chunk of multicluster allocations.

Reply:
404 means the client is asking gameserverallocation for a resource that is not available. It is a client error and should be handled by client. If a cluster is out of gameservers to allocate, then it returns 500. Does it make sense?

Reply:
Oh yeah totally :+1: I'm just thinking through worst case scenarios.

Like someone accidentally puts a different service in place of where the allocator used to be setup :fearful: 

Not incredibly likely, but just thinking through it.